crates/core/tedge_watchdog/src/dummy_watchdog.rs

```diff
@@ -0,0 +1,4 @@
+pub async fn start_watchdog(_config_dir: PathBuf) -> Result<(), anyhow::Error> {
+    // nothing here, as this is a dummy.
+    Ok(())
```

Comment:
Should this return Ok? Since a watchdog is generally seen as rather important, I don't think we should report success if we do not actually start it.

Reply:
So would you rather use a `Error::NotImplemented` or something like this?

Reply:
Something like that yes, or "NotAvailable" to be even clearer.